### PR TITLE
Allow users more flexibility when opening a DB

### LINKED ISSUE #383: Improve ease of selecting a database on the filesystem
https://github.com/github/codeql-coreql-team/issues/70

Slightly confusing that a database is a folder that can be navigated deeper into.

We could allow the user to select the (suitably renamed) `.dbinfo` file instead.

### PR DESCRIPTION
**Pull request by aeisenberg**
Closes #383.

See the heuristics in the issue.

### CODE REVIEW

File: extensions/ql-vscode/src/databases-ui.ts [hunk 1]
```diff
@@ -361,13 +362,34 @@ export class DatabaseUI extends DisposableObject {
...
+   *
+   * 1. If the selected URI to add is a file, choose the containing directory
+   * 2. If the selected URI is a directory matching db-*, choose the containing directory
+   * 3. choose the current directory
+   *
+   * @param uri
+   */
+  private async fixDbUri(uri: Uri): Promise<Uri> {
+    let dbPath = uri.fsPath;
+    if ((await fs.stat(dbPath)).isFile()) {
```
Comment (jcreedcmu) [inline extensions/ql-vscode/src/databases-ui.ts:387]: Does this break selecting a `.zip` archive of a db? If not, maybe refine the comment slightly to emphasize that we're only talking about the codepath that is concerned with intentionally selecting a database *directory*.

Comment (jcreedcmu) [reply extensions/ql-vscode/src/databases-ui.ts:387]: Yeah, oops, that definitely is the case here. Just requesting slightly more context in the doc comment then.

File: extensions/ql-vscode/src/databases-ui.ts [hunk 2]
```diff
@@ -361,13 +362,34 @@ export class DatabaseUI extends DisposableObject {
...
+      dbPath = path.dirname(dbPath);
+    }
+    if (path.basename(dbPath).startsWith('db-')) {
+      dbPath = path.dirname(dbPath);
+    }
+    return Uri.parse(`file:${dbPath}`, true);
```
Comment (jcreedcmu) [inline extensions/ql-vscode/src/databases-ui.ts:393]: I think there's a `Uri.file` method to construct a `Uri` out of a path

Comment (jcreedcmu) [reply extensions/ql-vscode/src/databases-ui.ts:393]: (Same applies to all the 
```
Uri.parse(`string interpolation`)
``` 
going on below)

File: extensions/ql-vscode/test/data/library.qll [hunk 3]
```diff
@@ -1,3 +1,3 @@
 predicate foo() {
- 1 == 1
-}
\ No newline at end of file
+ 1 = 1
```
Comment (jcreedcmu) [inline extensions/ql-vscode/test/data/library.qll:2]: hah, how long was this broken, I wonder

Comment (aeisenberg) [reply extensions/ql-vscode/test/data/library.qll:2]: Oh...didn't even mean to commit this. I'll extract to a separate commit.

**Comment by aeisenberg** [comment]
Added a changelog update, but I don't think this is worth pinging the documentation team.

**Pull request merged**